File: c/cz/mg/vulkan/PFN_vkCmdClearDepthStencilImage.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCmdClearDepthStencilImage_call(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong image, jlong imageLayout, jlong pDepthStencil, jlong rangeCount, jlong pRanges){
    (void)env;
    (void)jc;
    PFN_vkCmdClearDepthStencilImage f = (PFN_vkCmdClearDepthStencilImage)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkImage*)jniLongToPointer(image)),
        *((VkImageLayout*)jniLongToPointer(imageLayout)),
        ((VkClearDepthStencilValue*)jniLongToPointer(pDepthStencil)),
        *((uint32_t*)jniLongToPointer(rangeCount)),
        ((VkImageSubresourceRange*)jniLongToPointer(pRanges))
    );
}

File: cz/mg/vulkan/vk/PFNvkCmdClearDepthStencilImage.java
package cz.mg.vulkan.vk;

public class PFNvkCmdClearDepthStencilImage extends VkFunctionPointer {
    public PFNvkCmdClearDepthStencilImage() {
    }

    public PFNvkCmdClearDepthStencilImage(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCmdClearDepthStencilImage(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCmdClearDepthStencilImage(long value) {
        setValue(value);
    }

    public PFNvkCmdClearDepthStencilImage(VkInstance instance) {
        super(instance, new VkString("vkCmdClearDepthStencilImage"));
    }

    public void call(VkCommandBuffer commandBuffer, VkImage image, VkImageLayout imageLayout, VkClearDepthStencilValue pDepthStencil, VkUInt32 rangeCount, VkImageSubresourceRange pRanges){
        call(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.NULL_ADDRESS, image != null ? image.getVkAddress() : VkPointer.NULL_ADDRESS, imageLayout != null ? imageLayout.getVkAddress() : VkPointer.NULL_ADDRESS, pDepthStencil != null ? pDepthStencil.getVkAddress() : VkPointer.NULL, rangeCount != null ? rangeCount.getVkAddress() : VkPointer.NULL_ADDRESS, pRanges != null ? pRanges.getVkAddress() : VkPointer.NULL);
    }

    private static native void call(long vkaddress, long commandBuffer, long image, long imageLayout, long pDepthStencil, long rangeCount, long pRanges);
}
